classes/user/UserAction.inc.php

```diff
@@ -144,6 +144,14 @@ function mergeUsers($oldUserId, $newUserId) {
 			$giftDao->updateObject($gift);
 		}
 
+		// Transfer completed payments.
+		$paymentDao =& DAORegistry::getDAO('OJSCompletedPaymentDAO');
+		$paymentFactory = $paymentDao->getByUserId($oldUserId);
+		while ($payment =& $paymentFactory->next()) {
+			$payment->setUserId($newUserId);
+			$paymentDao->updateObject($payment);
```

Comment:
When looping using references, it's best to unset the the object at the end of the loop.